root/oauth2/Authorize.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import {
+  ACCESS_SCOPE_PROFILE,
+  ACCESS_SCOPE_EMAIL,
+  ACCESS_SCOPE_TAG,
+  ACCESS_SCOPE_RATING,
+  ACCESS_SCOPE_COLLECTION,
+  ACCESS_SCOPE_SUBMIT_ISRC,
+  ACCESS_SCOPE_SUBMIT_BARCODE,
+} from '../constants';
+import {withCatalystContext} from '../context';
+import Layout from '../layout';
+import {l, N_l} from '../static/scripts/common/i18n';
+
+const ACCESS_SCOPE_PERMISSIONS = {
+  [ACCESS_SCOPE_COLLECTION]: N_l('View and modify your private collections'),
+  [ACCESS_SCOPE_EMAIL]: N_l('View your email address'),
+  [ACCESS_SCOPE_PROFILE]: N_l('View your public account information'),
+  [ACCESS_SCOPE_RATING]: N_l('View and modify your private ratings'),
+  [ACCESS_SCOPE_SUBMIT_BARCODE]: N_l('Submit new barcodes to the database'),
+  [ACCESS_SCOPE_SUBMIT_ISRC]: N_l('Submit new ISRCs to the database'),
+  [ACCESS_SCOPE_TAG]: N_l('View and modify your private tags'),
+};
+
+type Props = {|
+  +$c: CatalystContextT,
+  +application: ApplicationT,
+  +offline: boolean,
+  +permissions: $ReadOnlyArray<number>,
+|};
+
+const Oauth2Authorize = withCatalystContext(({
```

Comment:
Is there any benefit to using withCatalystContext here instead in the export of as we usually do?

Reply:
Nope, since that way is more common I'll change it for consistency